### DISCUSSION TITLE
How to substite g** keymaps with \** keymaps

### DISCUSSION THREAD

**Discussion by filchristou** [numToStr/Comment.nvim#357] [Q&A]
I want to replace all gc[whatever] gb[whatever] with '\c[whatever]' and '\c[whatever]' correspondingly.
My Comment set up is:
```
require('Comment').setup{
 toggler = {
        ---Line-comment toggle keymap
        line = '\\cc',
        ---Block-comment toggle keymap
        block = '\\bc',
    },
    ---LHS of operator-pending mappings in NORMAL and VISUAL mode
    opleader = {
        ---Line-comment keymap
        line = '\\c',
        ---Block-comment keymap
        block = '\\b',
    },
    ---LHS of extra mappings
    extra = {
        ---Add comment on the line above
        above = '\\cO',
        ---Add comment on the line below
        below = '\\co',
        ---Add comment at the end of line
        eol = '\\cA',
    },
}
```

As a result now both the keymaps starting with `\` and the ones with `g` are enabled.
How to disable all the Comments.nvim keymaps that start with `g` in order to substitute them with `\` ?

**Comment by alexjegor** [comment]
I've also faced a similar problem, and neither redefining keymaps nor opting out of basic ones like this:
```
opts = {
  -- Disable default mappings
  mappings = {
    basic = false,
    extra = false,
  },
}
```
helped me. 

What did help was explicitly redefining the <leader>g* keymap right after the Comment.setup() call. 

So I ended up with the following config:
```
{
    'numToStr/Comment.nvim',
    opts = {
      -- Disable default mappings
      mappings = {
        basic = false,
        extra = false,
      },
    },
    config = function()
      local comment = require 'Comment'
      comment.setup()

      -- Redefine gb keymap right after Comment setup to avoid conflicts
      vim.keymap.set('n', 'gb', '<C-o>', { noremap = true, desc = 'Go [b]ack' })

      -- Custom keymaps
      vim.keymap.set('n', '<leader>/', function()
        require('Comment.api').toggle.linewise.current()
      end, { desc = 'Toggle comment' })
      vim.keymap.set(
        'v',
        '<leader>/',
        '<ESC><CMD>lua require("Comment.api").toggle.linewise(vim.fn.visualmode())<CR>',
        { desc = 'Toggle comment' }
      )
    end,
}
```

**Comment by Drayux** [comment]
The default gcc keymap is (frusteratingly) an oversight with the current state of neovim itself. It's created in the execution of the runtime defaults, which is ran *after* all of your initial user configuration. On most distros, this will be somewhere such as: `/usr/share/nvim/runtime/lua/vim/_defaults.lua`

The easiest way to counter act this is to:
a) Set your keybinds in the lua/after portion of your configuration (you can remap g to whatever here and it should be what you want
Or if you're like me and don't want to use the after directory for a single file,
b) Create a function that assigns your keybinds, and call it in an autocommand
```lua
vim.api.nvim_create_autocmd("VimEnter", {
    callback = function()
        vim.keymap.set("n", "gcc", ...)
        -- ...
    end
})
```
